EasyClangComplete.py

```diff
@@ -75,37 +76,38 @@ def __init__(self):
         logging.basicConfig(level=logging.DEBUG)
 
     def on_plugin_loaded(self):
-        """Called upon plugin load event."""
-        self.settings = plugin_settings.Settings()
+        """ Called upon plugin load event."""
+        self.settings_manager = SettingsManager()
         self.on_settings_changed()
-        self.settings.add_change_listener(self.on_settings_changed)
+        self.settings_manager.add_change_listener(self.on_settings_changed)
         # As the plugin have just loaded, we might have missed an activation
         # event for the active view so completion will not work for it until
         # re-activated. Force active view initialization in that case.
         self.on_activated_async(sublime.active_window().active_view())
 
     def on_settings_changed(self):
-        """Called when any of the settings changes."""
+        """ Called when any of the settings changes."""
         # If verbose flag is set then respect default DEBUG level.
         # Otherwise disable level DEBUG and allow INFO and higher levels.
-        off_level = logging.NOTSET if self.settings.verbose else logging.DEBUG
+        user_settings = self.settings_manager.user_settings()
+        off_level = logging.NOTSET if user_settings.verbose else logging.DEBUG
         logging.disable(level=off_level)
 
         # init everything else
         self.completer = None
-        if self.settings.use_libclang:
+        if user_settings.use_libclang:
```

Comment:
Maybe #119 should be kept open until that is fixed then.
Fixing that would also allow unit tests to use settings that are completely independent from user settings. Currently I often have to revert my user settings before I want to unit test as otherwise those would affect tests.